packages/nimble-components/src/table/specs/table-column-specs/table-column-anchor-hld.md

```diff
@@ -0,0 +1,121 @@
+
+To handle the event, we must find the associated anchor column directive. We have the `nimble-anchor` that the event came from, and by calling `anchor.getRootNode().host`, we can get the `nimble-table-cell`. Currently, there is no way to get from a given table cell to its associated column. One way to solve this is to include the associated `columnId` in the cell state, i.e. `cell.cellState.columnId`.
+```ts
+export interface TableCellState<...> {
+    cellRecord: TCellRecord;
+    columnConfig: TColumnConfig;
+    columnId: string | undefined;
+}
+```
+`columnId` is at the top level rather than part of the `columnConfig`, because it applies for all column types, and it is not for configuration.
+Adding `columnId` to the cell state would be a change to `nimble-components` just to support `nimble-angular`, but it seems generic enough to be a reasonable API change.
+
+One problem is that `columnId` is `undefined` unless a client sets it. We would also need to start auto-generating unique `columnId` values if the client hasn't provided one (i.e. in `connectedCallback`). An alternative would be to document that users must provide column ids when wishing to integrate with the Angular router, but this seems like an arbitrary requirement that would be better to avoid. Another option considered was to reuse the existing `internalUniqueId` that each column defines, but this is used for the action menu slot, and overloading its use as a general-purpose ID seems wrong.
+
+Another issue is that the client may change the `columnId` programmatically, invalidating the value stored in the cell state. To ensure consistency, we must have a `columnIdChanged()` handler that causes `refreshRows()` to be called on the table.
+
+Now that we have the `columnId` for a given anchor click, we need to get the column directive. We can get all potential directives as content children:
+```ts
+@ContentChildren(NimbleTableColumnAnchorDirective) public anchorColumnDirectives: QueryList<NimbleTableColumnAnchorDirective>;
+```
+Each directive has a reference to its associated element, so we can get to the `nimble-table-column-anchor`. From there, we have access to the `columnId`, so we can find the match.
+
+Finally, we need to call `RouterLinkWithHref.onClick()` to do the router navigation. However, we'll first need to set up the `routerLink` property with the url:
+```ts
+public doRouterNavigation(href: string): boolean {
+    this.routerLink = href;
+    return super.onClick(...);
+}
+```
+
+## Alternative Implementations / Designs
+
+Alternatives were described inline with proposed approaches.
+
+## Open Issues
```

Comment:
We should scan through recent table specs and the table column base class and see what other aspects of building a table column we'll need to specify. 

Some that come to mind:
1. will it be sortable? If so, which data field will it sort by? Any interesting comparison function needed?
2. will it be groupable? If so, which data field? And what does the group header look like? Is it a link too?
3. what column sizing modes will it support?
4. what are keyboard focus and keyboard link activation behaviors?
5. how does it look in the accessibility tree?
6. any Blazor considerations? Even if it's a standard wrapper, it would be good to call that out

Reply:
Maybe I should have used the spec template rather than an HLD, so I'm not sure what other parts are necessary. I've tried to address the specific items you've called out. I didn't really find much else to add, but if you notice anything else missing, I can add it.

Reply:
Thanks for the updates. I thought of one more case: Malcolm's recent cell state PR #1138. Is there any state that should be cleared when you start scrolling? The only thing I can think to consider would be the visuals for focus state. If I focus a link and then scroll, ideally the focus would not jump around to whatever other row is recycling that focused cell.

Reply:
Thanks, I hadn't thought about that. Yes, we need to clear the focus state, so I've added a note in the HLD about that.